test/e2e/flowaggregator_test.go

```diff
@@ -532,6 +505,105 @@ func testHelper(t *testing.T, data *TestData, podAIPs, podBIPs, podCIPs, podDIPs
+	if testOptions.enableCoverage {
+		antctlCovArgs := antctlCoverageArgs("antctl-coverage")
+		command = append(antctlCovArgs, args...)
+	} else {
+		command = append([]string{"antctl"}, args...)
+	}
+	t.Logf("Run command: %s", command)
+	stdout, stderr, err := runAntctl(podName, command, data)
+	if err != nil {
+		t.Fatalf("Error when running 'antctl get flowrecords -o json' from %s: %v\n%s", podName, err, antctlOutput(stdout, stderr))
+	}
+
+	var records []map[string]interface{}
+	err = json.Unmarshal([]byte(stdout), &records)
+	if err != nil {
+		t.Fatalf("Error when parsing flow records from antctl: %v", err)
+	}
+	if len(records) != 1 {
+		t.Fatalf("Number of flow records from antctl is not correct, epected 1 got %d", len(records))
+	}
+
+	checkAntctlRecord(t, records[0], srcIP, dstIP, srcPort, dstPort, isIPv6)
+}
+
+func checkAntctlRecord(t *testing.T, record map[string]interface{}, srcIP, dstIP, srcPort, dstPort string, isIPv6 bool) {
+	assert := assert.New(t)
+	if isIPv6 {
+		assert.Equalf(srcIP, record["sourceIPv6Address"], "The record from antctl does not have correct sourceIPv6Address")
+		assert.Equalf(dstIP, record["destinationIPv6Address"], "The record from antctl does not have correct destinationIPv6Address")
+	} else {
+		assert.Equalf(srcIP, record["sourceIPv4Address"], "The record from antctl does not have correct sourceIPv4Address")
+		assert.Equalf(dstIP, record["destinationIPv4Address"], "The record from antctl does not have correct destinationIPv4Address")
+	}
+	assert.Equalf(srcPort, fmt.Sprintf("%v", record["sourceTransportPort"]), "The record from antctl does not have correct sourceTransportPort")
+	assert.Equalf("perftest-a", record["sourcePodName"], "The record from antctl does not have correct sourcePodName")
+	assert.Equalf("antrea-test", record["sourcePodNamespace"], "The record from antctl does not have correct sourcePodNamespace")
+	assert.Equalf(controlPlaneNodeName(), record["sourceNodeName"], "The record from antctl does not have correct sourceNodeName")
+
+	assert.Equalf(dstPort, fmt.Sprintf("%v", record["destinationTransportPort"]), "The record from antctl does not have correct destinationTransportPort")
+	assert.Equalf("perftest-b", record["destinationPodName"], "The record from antctl does not have correct destinationPodName")
+	assert.Equalf("antrea-test", record["destinationPodNamespace"], "The record from antctl does not have correct destinationPodNamespace")
+	assert.Equalf(controlPlaneNodeName(), record["destinationNodeName"], "The record from antctl does not have correct destinationNodeName")
+
+	assert.Equalf(fmt.Sprintf("%v", ipfixregistry.FlowTypeIntraNode), fmt.Sprintf("%v", record["flowType"]), "The record from antctl does not have correct flowType")
+	assert.Equalf(fmt.Sprintf("%v", protocolIdentifierTCP), fmt.Sprintf("%v", record["protocolIdentifier"]), "The record from antctl does not have correct protocolIdentifier")
```

Comment:
I wonder if you could use `assert.EqualValues` to avoid the calls to `fmt.Sprintf`?

Reply:
done